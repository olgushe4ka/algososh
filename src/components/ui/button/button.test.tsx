import { render, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe("Button", () => {
    it("renders a button with text", () => {
      const { container } = render(<Button text="Click me" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  
    it("renders a button without text", () => {
      const { container } = render(<Button />);
      expect(container.firstChild).toMatchSnapshot();
    });
  
    it("renders a disabled button", () => {
      const { container } = render(<Button disabled={true} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  
    it("calls the onClick function when the button is clicked", () => {
      const handleClick = jest.fn();
      const { getByText } = render(<Button text="Click me" onClick={handleClick} />);
      fireEvent.click(getByText("Click me"));
      expect(handleClick).toHaveBeenCalledTimes(1);
    });
  });