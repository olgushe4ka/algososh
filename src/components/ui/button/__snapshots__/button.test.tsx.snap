// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a button with text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    Click me
  </p>
</button>
`;

exports[`Button renders a button without text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button renders a disabled button 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;
